File: examples/python/search_field.py
class SearchField:
    def __init__(self):
        self.query = ""

    def on_input(self, text):
        self.query = text

    def render(self):
        return '''<div class="toolbar">
<input type="text" id="searchInput" placeholder="Search..." onkeydown="if(event.key==='Enter')doSearch()">
<button class="btn-search" onclick="doSearch()">Search</button>
<button class="btn-reset" onclick="location.href='/'">Reset</button>
</div>'''


class ButtonSearch:
    def __init__(self, table, field):
        self.table = table
        self.field = field
        self.results = []

    def on_click(self):
        self.results = self.table.search(self.field.query)

    def get_results(self):
        return list(self.results)

File: examples/c++/search_field.cpp
#include <string>
#include <vector>

class SearchField {
public:
    std::string query;
    void onInput(const std::string& text) { query = text; }
};

class ButtonSearch {
    DataTable& table_;
    SearchField& field_;
    std::vector<Record> results_;
public:
    ButtonSearch(DataTable& table, SearchField& field) : table_(table), field_(field) {}
    void onClick() {
        results_.clear();
        for (const auto& r : table_.findAll()) {
            for (const auto& [k, v] : r.fields) {
                if (v.find(field_.query) != std::string::npos) {
                    results_.push_back(r);
                    break;
                }
            }
        }
    }
    std::vector<Record> getResults() const { return results_; }
};
